x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/secrets/SecretsIndexService.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector.secrets;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.action.admin.indices.template.put.PutIndexTemplateRequest;
+import org.elasticsearch.indices.SystemIndexDescriptor;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.core.template.TemplateUtils;
+
+import java.util.Map;
+
+import static org.elasticsearch.xpack.core.ClientHelper.CONNECTORS_ORIGIN;
+
+/**
+ * A service that manages persistent Connector Secrets.
+ */
+public class SecretsIndexService {
+
+    public static final String CONNECTOR_SECRETS_INDEX_NAME = ".connector-secrets";
+
+    private static final int CURRENT_INDEX_VERSION = 1;
+    private static final String MAPPING_VERSION_VARIABLE = "connector-secrets.version";
+    private static final String MAPPING_MANAGED_VERSION_VARIABLE = "connector-secrets.managed.index.version";
+
+    /**
+     * Returns the {@link SystemIndexDescriptor} for the Connector Secrets system index.
+     *
+     * @return The {@link SystemIndexDescriptor} for the Connector Secrets system index.
+     */
+    public static SystemIndexDescriptor getSystemIndexDescriptor() {
+        PutIndexTemplateRequest request = new PutIndexTemplateRequest();
+
+        String templateSource = TemplateUtils.loadTemplate(
+            "/connector-secrets.json",
+            Version.CURRENT.toString(),
+            MAPPING_VERSION_VARIABLE,
+            Map.of(MAPPING_MANAGED_VERSION_VARIABLE, Integer.toString(CURRENT_INDEX_VERSION))
+        );
+        request.source(templateSource, XContentType.JSON);
+
+        return SystemIndexDescriptor.builder()
+            .setIndexPattern(CONNECTOR_SECRETS_INDEX_NAME + "*")
+            .setPrimaryIndex(CONNECTOR_SECRETS_INDEX_NAME + "-" + CURRENT_INDEX_VERSION)
+            .setDescription("Secret values managed by Connectors")
+            .setMappings(request.mappings())
+            .setSettings(request.settings())
```

Comment:
Question, do we want specific settings here? For example, some of our other enterprise search system indices set the number of shards/replicas, and refresh interval. 

Note: If you decide to set a refresh interval below what serverless supports you'll need to update serverless to use fast refresh.

Reply:
I think explicitly defining settings is probably a good thing. I've added the default that the other system indices use.
Regarding fast refresh -- how would I determine if this index requires < 5s refresh interval? I'm not sure what the repercussions are here in either direction.

Reply:
This came up in another comment as well; it appears Fleet uses fast refresh for their secrets storage so that's a good precedent for us to copy them.
https://github.com/elastic/elasticsearch-serverless/blob/a40125757d2c60fc8b9b352d3e235d54cb362d9d/modules/serverless-fleet/src/main/java/co/elastic/elasticsearch/serverless/fleet/ServerlessFleetPlugin.java#L38

I'll open a followup PR to make that change. Thanks!